### PR TITLE
Return status code from HTTP module

### PR DESCRIPTION
**Pull request by Siegrift**
Closes https://github.com/api3dao/commons/issues/69

### CODE REVIEW

File: src/blockchain-utilities/derivation.ts [hunk 1]
```diff
@@ -55,3 +55,25 @@ export const deriveBeaconId = (airnodeAddress: Address, templateId: Hex) =>
 export const deriveBeaconSetId = (beaconIds: Hex[]) =>
   // By convention beacon IDs are sorted alphabetically - the ordering impacts the resulting hash.
   ethers.utils.keccak256(ethers.utils.defaultAbiCoder.encode(['bytes32[]'], [beaconIds]));
+
+export const verifyAirnodeXpub = (airnodeXpub: string, airnodeAddress: Address): ethers.utils.HDNode => {
```
Comment (andreogle) [inline src/blockchain-utilities/derivation.ts:59]: I would kinda expect a function named "verify" to return a boolean. That said, I don't really have a better suggestion at the moment 🤷🏻‍♂️

File: src/blockchain-utilities/derivation.ts [hunk 2]
```diff
@@ -55,3 +55,25 @@ export const deriveBeaconId = (airnodeAddress: Address, templateId: Hex) =>
...
+  // so we must derive the child default derivation path m/44'/60'/0'/0/0
+  // to compare it and check if xpub belongs to the Airnode wallet
+  const hdNode = ethers.utils.HDNode.fromExtendedKey(airnodeXpub);
+  if (airnodeAddress !== hdNode.derivePath('0/0').address) {
+    throw new Error(`xpub does not belong to Airnode: ${airnodeAddress}`);
+  }
+  return hdNode;
+};
+
+export function deriveSponsorWalletAddress(
```
Comment (andreogle) [inline src/blockchain-utilities/derivation.ts:70]: What's with the `export function` and `export const verifyAirnodeXpub` syntax differences?

**Review by andreogle — APPROVED** [review]

File: src/blockchain-utilities/derivation.ts (hunk 1 above)
Comment (Siegrift) [reply src/blockchain-utilities/derivation.ts:59]: Yeah, but you kinda want to return the HD node. This is copied over from Airnode which is another point for preserving this as is (people are familiar and need to migrate less).

**Comment by Siegrift** [comment]
Also, most of these changes come from https://github.com/api3dao/commons/pull/68 because the PR was not rebased.

**Pull request merged**